commit 3904414a3008751ffcaf18fbe33bd34812b7bfe5
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Mon May 30 01:24:09 2016 +0200

    hurd: fix _hurd_self_sigstate reference from ____longjmp_chk
    
    	* sysdeps/mach/hurd/i386/____longjmp_chk.S (____longjmp_chk) [PIC]:
    	  Use PLT entry for calling _hurd_self_sigstate.

diff --git a/sysdeps/mach/hurd/i386/____longjmp_chk.S b/sysdeps/mach/hurd/i386/____longjmp_chk.S
index 4ddf6cd..dec8ea7 100644
--- a/sysdeps/mach/hurd/i386/____longjmp_chk.S
+++ b/sysdeps/mach/hurd/i386/____longjmp_chk.S
@@ -66,7 +66,12 @@ ENTRY (____longjmp_chk)
 
 	/* TODO: need locking?  */
 	/* struct hurd_sigstate * _hurd_self_sigstate (void) */
-	call	_hurd_self_sigstate
+#ifdef PIC
+	call    1f
+1:	popl    %ebx
+	addl    $_GLOBAL_OFFSET_TABLE_+[.-1b], %ebx
+#endif
+	call	JUMPTARGET(_hurd_self_sigstate)
 	/* TODO: %eax and %eax->sigaltstack are always valid?  */
 
 	testl	$SS_ONSTACK, (HURD_SIGSTATE__SIGALTSTACK__OFFSET + SIGALTSTACK__SS_FLAGS__OFFSET)(%eax)
